framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1242,15 +1242,19 @@ public AnnotationMirror createIntRangeAnnotation(long from, long to) {
     }
 
     /**
-     * Create an {@code @IntRange} annotation from the range. May return BOTTOMVAL or UNKNOWNVAL.
+     * Create an {@code @IntRange} or {@code @IntVal} annotation from the range. May return
+     * BOTTOMVAL or UNKNOWNVAL.
      */
     public AnnotationMirror createIntRangeAnnotation(Range range) {
         if (range.isNothing()) {
             return BOTTOMVAL;
         } else if (range.isEverything()) {
             return UNKNOWNVAL;
-        } else {
+        } else if (range.isWiderThan(MAX_VALUES)) {
             return createIntRangeAnnotation(range.from, range.to);
+        } else {
```

Comment:
I was motivated by the example when the user writes `@IntRange(from = 0, to = 1)` and is shown instead the more verbose `@IntVal({0, 1, 2, 3, 4, 5, 6, 7, 8, 9})`.  I thought that consistency with what the user wrote was desirable.

I agree with Jason that we should be consistent regardless.